sdk/go/pulumi/context.go

```diff
@@ -177,15 +183,26 @@ func (ctx *Context) Context() context.Context {
 	return ctx.ctx
 }
 
+// Returns a copy of base context in which the value associated with key is val.
+func (ctx *Context) WithValue(key, val any) *Context {
+	newCtx := &Context{
+		ctx:   ctx.ctx,
+		state: ctx.state,
```

Comment:
I was wondering why we were separating the `state` out here, but given that we're copying the context, but we want to have the same state everywhere I think this makes sense.